src/libraries/System.Net.Security/src/System/Net/Security/MD4.cs

```diff
@@ -0,0 +1,235 @@
+            /* Update number of bits */
+            count[0] += (uint)(cbSize << 3);
+            if (count[0] < (cbSize << 3))
+                count[1]++;
+            count[1] += (uint)(cbSize >> 29);
+
+            int partLen = 64 - index;
+            int i = 0;
+            /* Transform as many times as possible. */
+            if (cbSize >= partLen)
+            {
+                BlockCopy(input, ibStart, buffer, index, partLen);
+                MD4Transform(state, buffer, 0);
+
+                for (i = partLen; i + 63 < cbSize; i += 64)
+                {
+                    MD4Transform(state, input, ibStart + i);
+                }
+
+                index = 0;
+            }
+
+            /* Buffer remaining input */
+            BlockCopy(input, ibStart + i, buffer, index, (cbSize - i));
+        }
+
+        //--- private methods ---------------------------------------------------
+
+        private static void BlockCopy(ReadOnlySpan<byte> source, int srcOffset, Span<byte> destination, int dstOffset, int count)
+        {
+            for (int srcIndex = srcOffset, dstIndex = dstOffset; srcIndex < srcOffset + count; srcIndex++, dstIndex++)
+            {
+                destination[dstIndex] = source[srcIndex];
+            }
+        }
+
+        /* F, G and H are basic MD4 functions. */
+        private static uint F(uint x, uint y, uint z)
+        {
+            return (uint)(((x) & (y)) | ((~x) & (z)));
+        }
+
+        private static uint G(uint x, uint y, uint z)
+        {
+            return (uint)(((x) & (y)) | ((x) & (z)) | ((y) & (z)));
+        }
+
+        private static uint H(uint x, uint y, uint z)
+        {
+            return (uint)((x) ^ (y) ^ (z));
+        }
+
+        /* ROTATE_LEFT rotates x left n bits. */
+        private static uint ROL(uint x, byte n)
+        {
+            return (uint)(((x) << (n)) | ((x) >> (32 - (n))));
+        }
+
+        /* FF, GG and HH are transformations for rounds 1, 2 and 3 */
+        /* Rotation is separate from addition to prevent recomputation */
+        private static void FF(ref uint a, uint b, uint c, uint d, uint x, byte s)
+        {
+            a += F(b, c, d) + x;
+            a = ROL(a, s);
+        }
+
+        private static void GG(ref uint a, uint b, uint c, uint d, uint x, byte s)
+        {
+            a += G(b, c, d) + x + 0x5a827999;
+            a = ROL(a, s);
+        }
+
+        private static void HH(ref uint a, uint b, uint c, uint d, uint x, byte s)
+        {
+            a += H(b, c, d) + x + 0x6ed9eba1;
+            a = ROL(a, s);
+        }
+
+        private static void Encode(Span<byte> output, Span<uint> input)
+        {
+            for (int i = 0, j = 0; j < output.Length; i++, j += 4)
+            {
+                BinaryPrimitives.WriteUInt32LittleEndian(output.Slice(j), input[i]);
+            }
+        }
+
+        private static void Decode(Span<uint> output, ReadOnlySpan<byte> input, int index)
```

Comment:
As with other span-accepting methods, you can eliminate the index parameter and just update the caller to slice the span to pass the correct start.